### DISCUSSION TITLE
Trackable detecion - PlaneRenderer

### DISCUSSION THREAD

**Discussion by SantiCamm** [sceneview/sceneview#537] [Q&A]
Hi! Hope you are doing great.
I'm making a small measurement project, in which I added like a "pointer" from which the users make their measurements.
I depend on the trackable/planes detection from all the surfaces that the camera points at, normal usage of ARCore.
I wanted to know if it was possible to make something like this:
![image](https://github.com/SceneView/sceneview-android/assets/32301876/39088d30-d9e1-426a-9875-5dd886bcae71)

I mean, the area on which the surfaces are being detected, reduced to some extend near the pointer (center of the screen, for instance).

Thanks in advance!

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, plane visualization and detection is configured through `ARSceneView`:

```kotlin
ARSceneView(
    sessionConfiguration = { session, config ->
        config.planeFindingMode = Config.PlaneFindingMode.HORIZONTAL_AND_VERTICAL
    },
    onSessionUpdated = { session, frame ->
        val planes = session.getAllTrackables(Plane::class.java)
        for (plane in planes) {
            if (plane.trackingState == TrackingState.TRACKING) {
                // Detected plane — create anchor, show indicator, etc.
            }
        }
    }
)
```

Plane rendering (the grid overlay) is handled by ARSceneView. You can access the detected trackables in `onSessionUpdated` to create anchors or show custom indicators.
